docs/codesigning-for-apple-developers.md

```diff
@@ -0,0 +1,127 @@
+# Codesigning for Apple Developers
+
+With the introduction of Apple's EndpointSecurity framework ([PR
+6467](https://github.com/osquery/osquery/pull/6467), we've started to
+need entitlements, profiles, and codesigning. Even in development.
+
+As I understand Apple's model:
+1. apps/packages/etc are signed by certs. 
+2. There are _many_ different flavors of certificate. For our
+   purposes, the ones that matter:
+   - Developer ID Installer: Privileged. Allows installer distribution
+     outside the app store.
+   - Developer ID Application: Privileged. Allows app distribution
+     outside the app store.
+   - Apple Development / Mac Development: Developer. Allows signing
+     development versions.
+3. The privileged ones, coupled with an entitlements file allow
+   osquery to work.
+4. The developer ones _also_ require a profile, which only works on
+   macs identified by uuid.
+   - `select uuid from system_info`
+5. Apple accounts can have members that are `developers`. These
+   members can create personal developer certificates, but cannot
+   create distribution ones. Cannot adjust devices. Cannot adjust
+   profiles. According, it should be pretty reasonable to add osquery
+   developers to the account.
+   
+## Process for adding a new developer
+
+_*Note*: This is for adding a new developer to the account. This
+should not grant distribution permission, but they should still be somewhat trusted._
+
+Create a foundation ticket for general tracking and details. If any of
+this is considered private, consider using slack to communicate
+details.
+
+We will need the email address for their apple account. They can
+create a new one, or use an existing one. This should be a
+human. Apple does not like roll accounts.
```

Comment:
I think you mean `role` here

Reply:
I don't know what Apple likes or doesn't like, but FWIW, at the Trail of Bits team we have an account / "person" called Continuous Integration, which signs in CI for testing (has its own Development certificate, is added to the Development provisioning profile).